### DISCUSSION TITLE
Listen to location changes of window/browser vs fetch requests

### DISCUSSION THREAD

**Discussion by karfau** [cypress-io/cypress#14854] [Questions and Help] [answered]
In my test I want to assert that requests to certain URLs are done via `fetch` and not as a location change (e.g. by setting `window.location`).
When intercepting the request it's not obvious to me how I can tell them apart.
- I could imagine that I would be able to solve my task if I would be able to "add a listener for some event that is triggered when the location changes" but I couldn't find any related information.
- I tried to `intercept` all related requests and make assertions on them but it seems as if this specific multi level redirect that is triggered (and correctly followed) by a fetch request can not be intercepted. I only get the first redirect and even seeing the other request happening in the browser I need to intercept them separately which is quite tedious and error prone.
- In the cypress ui (using `cypress open`) I can see the difference being indicated by ***`(new url)`*** and ***`(req)`***:
![image](https://user-images.githubusercontent.com/135657/106526726-671ad700-64e6-11eb-9dff-0a8f46d34403.png)
but have can I test for this difference?

**Comment by karfau** [comment, marked as answer]
I found one solution (that currently only works when running the tests in a modern chromium based browser):
https://www.w3.org/TR/fetch-metadata/#sec-fetch-mode-header

But are there others?

**Reply by bahmutov** [reply]
I'm not sure I understand what you are trying to do. Are you trying to confirm the url of the page? You can use cy.location command and assertion for this. Do you want to spy on fetch requests? Use cy.intercept for this

**Reply by karfau** [reply]
I'm trying to assert that certain URLs are **not** "part of the browser history" but the call happens via `fetch` and follows the redirected before putting the resulting URL into the browser.

**Comment by bahmutov** [comment]
Can you create an app or reproducible example showing what you mean in action?

**Reply by karfau** [reply]
Thx for the question and the fast reply!

The more I think about it ... I guess one of the reasons why it's so complicated is that some part of my use case is sooo legacy and we can't touch it. And the whole setup is quite complicated because of that.

Not sure that I will be able to provide a simplified example any time soonish, since I'm happy with the solution I found (linked above).
